src/main/java/com/mitchej123/hodgepodge/core/HodgepodgeMixinPlugin.java

```diff
@@ -58,189 +40,68 @@ public void acceptTargets(Set<String> myTargets, Set<String> otherTargets) {
 
     @Override
     public List<String> getMixins() {
-        List<String> mixins = new ArrayList<>();
-        for (MixinSets mixin : MixinSets.values()) {
-            if (mixin.shouldBeLoaded() && mixin.loadJar()) {
-                mixins.addAll(mixin.mixinClasses);
-                log.info("Loading hodgepodge plugin '{}' with mixins: {}", mixin.name, mixin.mixinClasses);
+        final boolean isDevelopmentEnvironment = (boolean) Launch.blackboard.get("fml.deobfuscatedEnvironment");
+
+        List<TargetedMod> loadedMods = Arrays.stream(TargetedMod.values())
+            .filter(mod -> (
+                        mod == TargetedMod.VANILLA 
+                            || (isDevelopmentEnvironment && MinecraftURLClassPath.findJarInClassPath(mod.devJarName))
+                            || loadModJar(mod)
+                    )
+            )
+            .collect(Collectors.toList());
+
+        for (TargetedMod mod : TargetedMod.values()) {
+            if(loadedMods.contains(mod)) {
+                Hodgepodge.log.info("Found Mod " + mod.modName + "!");
+            }
+            else if (ArrayUtils.contains(Hodgepodge.config.requiredMods, mod.modName)){
+                Hodgepodge.log.error("CRITICAL ERROR: Could not find required jar {}.  If this mod is not required please remove it from the 'requiredMods' section of the config.", mod.modName);
+                FMLCommonHandler.instance().exitJava(-1, true);
```

Comment:
@SinTh0r4s Adding optional hard deps for mixins is something I'd recommend upstreaming.  (Optional) Hard failure when the target mod isn't found, so we don't have mixins silently failing due to a mod rename (pam's harvest vs pam's_harvest)